File: observer/ShippingCompany.java
package observer;

public class ShippingCompany implements Subscriber {
    private String name;

    public ShippingCompany(String name) {
        this.name = name;
    }

    @Override
    public void notify(String message) {
        System.out.println(name + " is receiving notification about: " + message);
    }
}

File: typescript/observer/shipping-company.ts
import { Subscriber } from "./subscriber";

export class ShippingCompany implements Subscriber {
  constructor(private name: string) {}

  getName() {
    return this.name;
  }

  notify(message: string) {
    console.log(`${this.name} is receiving a notification about: ${message}`);
  }
}
